src/plugins/Transloadit/index.js

```diff
@@ -180,23 +184,19 @@ module.exports = class Transloadit extends Plugin {
           endpoint: assembly.tus_url
         })
 
-        // Set uppy server location.
-        // we only add this, if 'file' has the attribute remote, because
-        // this is the criteria to identify remote files. If we add it without
-        // the check, then the file automatically becomes a remote file.
-        // @TODO: this is quite hacky. Please fix this later
-        let remote
-        if (file.remote) {
+        // Set uppy server location. We only add this, if 'file' has the attribute
+        // remote, because this is the criteria to identify remote files.
+        // We only replace the hostname for Transloadit's uppy-servers, so that
+        // people can self-host them while still using Transloadit for encoding.
+        let remote = file.remote
+        if (file.remote && TL_UPPY_SERVER.test(file.remote)) {
```

Comment:
Trying to make sense of this in my head. If you are using Transloadit and your own Uppy Server, where does the Transloadit’s Uppy Server come from in the first place? Transloadit plugin forces it, and here we are overwriting it back? This is also tied to `service` option, which is basically “you might want to decide which API endpoint to use yourself, forgoing the load balancer”. I apologize in advance for asking this :)

Reply:
Previously, the Transloadit plugin always overrode the Uppy Server URL. With this change, it only sets the Uppy Server URL if you configured your provider plugins to use Transloadit's Uppy Server. If you are using your own Uppy Server, this change makes sure the Transloadit plugin doesn't touch it.